src/Send/Context/SendContext.tsx

```diff
@@ -1,35 +1,72 @@
 import * as React from 'react'
-import {createContext, useContext} from 'react'
 
 import {getDefaultAssetByNetworkId} from '../../legacy/config'
-import {isEmptyString} from '../../legacy/utils'
 import {YoroiWallet} from '../../yoroi-wallets'
-import {TokenId, YoroiAmounts} from '../../yoroi-wallets/types'
+import {TokenId} from '../../yoroi-wallets/types'
 
 type SendProvider = {
   wallet: YoroiWallet
-  balances: YoroiAmounts
 }
 
-const SendContext = createContext<undefined | SendContext>(undefined)
-export const SendProvider: React.FC<SendProvider> = ({children, balances, wallet}) => {
+const SendContext = React.createContext<undefined | SendContext>(undefined)
+export const SendProvider: React.FC<SendProvider> = ({children, wallet}) => {
   const primaryTokenId = getDefaultAssetByNetworkId(wallet.networkId).identifier
 
-  const [state, dispatch] = React.useReducer(sendReducer, {...initialState, selectedTokenId: primaryTokenId})
-  const sendActions = sendActionsMapper(dispatch)
-
-  React.useEffect(() => {
-    if (primaryTokenId !== state.selectedTokenId && isEmptyString(balances[state.selectedTokenId])) {
-      sendActions.setSelectedTokenId(primaryTokenId)
-      sendActions.clear()
-    }
-  }, [primaryTokenId, balances, sendActions, state.selectedTokenId, wallet.networkId])
+  const [state, dispatch] = React.useReducer(sendReducer, {
+    ...initialState,
+    primaryTokenId,
+    selectedTokenId: primaryTokenId,
+  })
 
   return (
     <SendContext.Provider
       value={{
-        sendActions,
-        sendState: state,
+        ...state,
```

Comment:
this will going cause unnecessary rerenders

Reply:
suggestion?

Reply:
`useRef`

Reply:
Done